### PR TITLE
Refactor client to have a single entrypoint

### PR DESCRIPTION
**Pull request by pdalpra**

### CODE REVIEW

File: src/main/scala/tv/teads/github/api/GithubApiClientBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package tv.teads.github.api
+
+private[api] case class GithubApiClientConfigBuilder(
+  userOrOrg:    Option[String] = None,
```
Comment (gbougeard) [inline src/main/scala/tv/teads/github/api/GithubApiClientBuilder.scala]: `owner` ?

Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/GithubApiClientBuilder.scala]: Sounds good :)

**Comment by gbougeard** [comment]
otherwise LGTM

Let's refactor!

**Comment by pdalpra** [comment]
And that's not the last PR, got a few other ones in the pipe :)

**Pull request merged**